### PR TITLE
Update for new fix structure

### LINKED ISSUE #687: Update for new 'fixed' directories
The 'fixed' directories are undergoing a reorganization. This will require several script changes. For details, see:

https://github.com/NOAA-EMC/global-workflow/issues/966#issuecomment-1227692287

### PR DESCRIPTION
**Pull request by WalterKolczynski-NOAA**
## DESCRIPTION OF CHANGES: 
Global workflow recently updated their fix directory structure to incorporate versioning using dates within the directory tree. Additionally, some files were moved and directories renamed (including removing all the 'fix_' prefixes of subdirectories). This commit updates the fix paths in UFS utils to match the new directory structure, and also take version numbers for each of the fix component directories, similar to how versions are used for modulefiles.)

## TESTS CONDUCTED: 
No new unit tests required.

- [x] cmake tests all pass on Hera for Intel
- [x] cmake tests all pass on Hera for gnu
- [x] cmake tests all pass on Orion
- [x] cmake tests all pass on Cactus
- [x] CI tests all pass
- [x] Global-workflow cycled test on Hera
- [x] Global-workflow coupled test on Hera

## DEPENDENCIES:
None

## DOCUMENTATION:
None needed

## ISSUE: 
Fixes #687
Refs: NOAA-EMC/global-workflow#966

## CONTRIBUTORS (optional): 
@KateFriedman-NOAA did a lot of the work before I finished it off and tested.

### CODE REVIEW

**Comment by WalterKolczynski-NOAA** [comment]
Kate didn't open a PR before she left, but I found her stuff and finished it off.

I don't have access to Jet to test there. I do not know who maintains fix on the other platforms not already tested or if they have been updated.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA S4 has the updated fix directories.  I will run a cycled and S2S test on that system.

I can run the UFS_UTILS regression tests on Jet, but the global workflow port has not been updated since March, so I won't be able to run a cycled or coupled test at this time.  Would you like me to go ahead with the RTs?

**Comment by GeorgeGayno-NOAA** [comment]
@WalterKolczynski-NOAA I can run the UFS_UTILS regression tests.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA gdas_init is looking for the mosaic files in the wrong location.  The chgres scripts all set `FIX_ORO=${FIX_FV3}/fix_fv3_gmted2010` instead of `${FIX_FV3}/orog`.

**Comment by WalterKolczynski-NOAA** [comment]
@DavidHuber-NOAA You won't be able to test workflow because I haven't opened a PR with the changes there. It is waiting on this PR before I open it. I don't think another workflow test is necessary for this PR.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA OK, that makes sense, thanks.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA I will run RTs on S4 instead.

**Comment by GeorgeGayno-NOAA** [comment]
> @WalterKolczynski-NOAA gdas_init is looking for the mosaic files in the wrong location. The chgres scripts all set `FIX_ORO=${FIX_FV3}/fix_fv3_gmted2010` instead of `${FIX_FV3}/orog`.

Good catch.

**Comment by GeorgeGayno-NOAA** [comment]
The `snow2mdl`, `chgres_cube` and `global_cycle` regression tests have old 'fix' paths.

**Review by GeorgeGayno-NOAA — CHANGES_REQUESTED** [review]
And two paths in this script need to be updated - ./util/sfc_climo_gen/run.wcoss2.sh.

**Comment by WalterKolczynski-NOAA** [comment]
Updated a bunch more paths that I missed the first time. The remaining paths weren't changed because they point to a static path, either for FTP/AWS, or a personal location not controlled by workflow.

```
hfe06 UFS_UTILS/ > egrep -RI --exclude-dir=fix '(fix_am)|(fix_fv3_gmted)|(fix_orog)|(fix_sfc_climo)'                                                                                                                                
docs/source/ufs_utils.rst:             * landcover30.fixed (unformatted binary). Located here `./fix/fix_orog <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_orog/>`_.
docs/source/ufs_utils.rst:             * gmted2010.30sec.int (unformatted binary). Located here `./fix/fix_orog <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_orog/>`_.
docs/source/ufs_utils.rst:             * thirty.second.antarctic.new.bin (unformatted binary). Located here `./fix/fix_orog <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_orog/>`_.
docs/source/ufs_utils.rst:      * geo_em.d01.lat-lon.2.5m.HGT_M.nc - global topographic data on 2.5-minute lat-lon grid (interpolated from GMTED2010 30-second topographic data). `Located here <https://noaa-ufs-srw-pds.s3.amazona
ws.com/index.html#fix/fix_am/>`_.
docs/source/ufs_utils.rst:      * HGT.Beljaars_filtered.lat-lon.30s_res.nc - global topographic data on 30-second lat-lon grid (GMTED2010 data smoothed according to Beljaars et al. (QJRMS, 2004)). `Located here <https://noaa-ufs
-srw-pds.s3.amazonaws.com/index.html#fix/fix_am/>`_.
docs/source/ufs_utils.rst:      * lake status code file - GlobalLakeStatus.dat (located in `./fix/fix_orog <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_orog/>`_). See GlobalLakeStatus.txt for the defintion of each code.
docs/source/ufs_utils.rst:      * lake depth file - GlobalLakeDepth.dat (located in `./fix/fix_orog <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_orog/>`_). See GlobalLakeDepth.txt for a description of this file.
docs/source/ufs_utils.rst:The surface climatological data is located here `./fix/fix_sfc_climo <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_sfc_climo/>`_.  All NetCDF.
docs/source/chgres_cube.rst:Users may create their own global grids, or use the pre-defined files are located `here <https://ftp.emc.ncep.noaa.gov/EIB/UFS/global/fix/fix_fv3_gmted2010.v20191213/>`_.
docs/source/chgres_cube.rst:      * FV3 surface climatological files - Located under the `./fix_sfc <https://ftp.emc.ncep.noaa.gov/EIB/UFS/global/fix/fix_fv3_gmted2010.v20191213/C48/fix_sfc>`_ sub-directory.  One file for each tile.  NetCDF format.
docs/source/chgres_cube.rst:      * FV3 vertical coordinate file.  Text file.  `Located here <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_am/>`_.
docs/source/chgres_cube.rst:      * FV3 vertical coordinate file.  Text file.  `Located here <https://noaa-ufs-srw-pds.s3.amazonaws.com/index.html#fix/fix_am/>`_.
tests/chgres_cube/data/config_spectral_sigio.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
tests/chgres_cube/data/config_fv3_tiled_warm_restart.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
tests/chgres_cube/data/config_fv3_tiled.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
tests/chgres_cube/data/config_fv3_netcdf.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
tests/chgres_cube/data/config_gaussian_nemsio.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
tests/chgres_cube/data/config_gfs_grib2.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l65.txt"
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 10
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 40
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 70
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 80
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt") stop 120
tests/chgres_cube/ftst_program_setup.F90:  if (trim(vcoord_file_target_grid) .ne. "/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l65.txt") stop 100
tests/sfc_climo_gen/CMakeLists.txt:set(SFCGEN2_URL "https://ftp.emc.ncep.noaa.gov/static_files/public/UFS/GFS/fix/fix_sfc_climo")
ush/sfc_climo_gen.sh:FIX_FV3=${FIX_FV3:-/scratch4/NCEPDEV/global/save/glopara/git/fv3gfs/fix/fix_fv3_gmted2010/C${res}}
build/tests/chgres_cube/fort.41:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
build/tests/chgres_cube/data/config_spectral_sigio.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
build/tests/chgres_cube/data/config_fv3_tiled_warm_restart.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
build/tests/chgres_cube/data/config_fv3_tiled.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
build/tests/chgres_cube/data/config_gaussian_nemsio.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l64.txt"
build/tests/chgres_cube/data/config_gfs_grib2.nml:  vcoord_file_target_grid="/scratch1/NCEPDEV/da/George.Gayno/ufs_utils.git/UFS_UTILS/reg_tests/chgres_cube/../../fix/fix_am/global_hyblev.l65.txt"
```

File: ush/chgres_cube.sh [hunk 1]
```diff
@@ -33,9 +33,9 @@ NWROOT=${NWROOT:-/nw${envir}}
 HOMEufs=${HOMEufs:-${NWROOT}/ufs_util.${ufs_ver}}
 EXECufs=${EXECufs:-$HOMEufs/exec}
 FIXufs=${FIXufs:-$HOMEufs/fix}
-FIXfv3=${FIXfv3:-$FIXufs/fix_fv3_gmted2010/C${CRES}}
-FIXsfc=${FIXsfc:-$FIXfv3/fix_sfc}
-FIXam=${FIXam:-$FIXufs/fix_am}
+FIXfv3=${FIXfv3:-$FIXufs/orog/C${CRES}}
+FIXsfc=${FIXsfc:-$FIXfv3/sfc}
```
Comment (GeorgeGayno-NOAA) [inline ush/chgres_cube.sh]: This is a different FIXsfc directory. It needs to be reverted to $FIXfv3/fix_sfc.

**Comment by WalterKolczynski-NOAA** [comment]
@GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:

1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.

Should I:
- Add them to this PR?
- Make a separate PR(s)?
- Just point you towards them?
- Ignore them?

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:
> 
> 1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
> 2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.
> 
> Should I:
> 
> * Add them to this PR?
> * Make a separate PR(s)?
> * Just point you towards them?
> * Ignore them?

Let's add them under a separate issue/PR. I tried to create a mpiexec.wcoss2 file one afternoon. I could not get it to work. Thanks for doing this.

**Comment by WalterKolczynski-NOAA** [comment]
> > @GeorgeGayno-NOAA I also have a couple local changes I made to do my testing, I don't know if you wanted me to add them to this PR:
> > 
> > 1. I created a `cmake/mpiexec.wcoss2` file and edited `cmake/LibMPI.cmake` to submit jobs on cactus.
> > 2. I increased the `pull_data` timeout of `tests/CmakeLists.txt` because the data download would fail with the shorter time.
> > 
> > Should I:
> > 
> > * Add them to this PR?
> > * Make a separate PR(s)?
> > * Just point you towards them?
> > * Ignore them?
> 
> Let's add them under a separate issue/PR. I tried to create a mpiexec.wcoss2 file one afternoon. I could not get it to work. Thanks for doing this.

Opened two new issues (#690 and #691). I already have the fixes, I just need to sort them into PRs.

**Review by JeffBeck-NOAA — APPROVED** [review]

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA The chgres_cube baselines store the surface fix files in `${HOMEreg}/fix/C${RES}/fix_sfc`, which are now inconsistent with the fix structure.  When the chgres_cube tests run, they are now looking in `sfc` instead of `fix_sfc`, which does not exist in the baseline input data.  This could be remedied by adding

`export FIXsfc=${FIXfv3}/fix_sfc`

to each of the chgres_cube test scripts (e.g. c96.gfs.sigio.sh).  Do you think this should happen or that the baseline surface fix directories should be renamed on each HPC?

**Comment by WalterKolczynski-NOAA** [comment]
@DavidHuber-NOAA I would recommend doing what we did when we made this change to fix: move the baseline to the new locations and temporarily set up symlinks pointing the old locations to the new, so older versions don't break.

**Comment by WalterKolczynski-NOAA** [comment]
Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.

**Comment by DavidHuber-NOAA** [comment]
@WalterKolczynski-NOAA OK, that sounds reasonable to me regarding the links, though I would defer to @GeorgeGayno-NOAA.  The regression tests are run via [rt.sh](https://github.com/ufs-community/UFS_UTILS/blob/develop/reg_tests/rt.sh).  That script needs to be modified first at the top by specifying a [working directory, account, etc](https://github.com/ufs-community/UFS_UTILS/blob/c33ea5242e7b2fa1c4199a8ff5b88523e371ba01/reg_tests/rt.sh#L5-L11).  Running `make test` runs the unit tests and not the regression tests.

**Comment by GeorgeGayno-NOAA** [comment]
> Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.

Only the chgres_cube regression tests are failing. If you make the fix to ./ush/chgres_cube.sh (see my comment) I can run them for you.

**Comment by WalterKolczynski-NOAA** [comment]
> > Also, are these different tests than I get when I run `make test`? If so, I wasn't able to figure out how to run them, otherwise I would have.
> 
> Only the chgres_cube regression tests are failing. If you make the fix to ./ush/chgres_cube.sh (see my comment) I can run them for you.

Which comment? I think I've already made the updates from [this comment](#issuecomment-1234575974). I think the remaining issues lie in the baseline location, which is what David and I were talking about yesterday.

File: ush/chgres_cube.sh (hunk 1 above)
Comment (GeorgeGayno-NOAA) [reply ush/chgres_cube.sh]: @WalterKolczynski-NOAA This is causing the chgres_cube regression tests to fail.

**Comment by WalterKolczynski-NOAA** [comment]
Is there anything you are waiting on from me that is holding this up?

**Comment by GeorgeGayno-NOAA** [comment]
> Is there anything you are waiting on from me that is holding this up?

Merge in the latest updates from 'develop' to your branch. And did you address the problem with the chgres_cube regression test script? See my review.

**Comment by GeorgeGayno-NOAA** [comment]
I can run the chgres test for you after you update the script.

File: fix/link_fixdirs.sh [hunk 2]
```diff
@@ -35,23 +35,28 @@ pwd=$(pwd -P)
 #--model fix fields
 #------------------------------
 if [ $machine = "hera" ]; then
-    FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix_NEW"
+    FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
 elif [ $machine = "jet" ]; then
     FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix_NEW"
```
Comment (GeorgeGayno-NOAA) [inline fix/link_fixdirs.sh]: @WalterKolczynski-NOAA the directory on Jet does not exist. Do you mean 'fix' instead of 'fix_NEW'?

**Comment by WalterKolczynski-NOAA** [comment]
I updated the branch. I think the issue your talking about are resolve, but see my earlier comment asking which comment you are referring to and the discussion with David about whether the baseline fix files need to be changed to match.

**Comment by GeorgeGayno-NOAA** [comment]
@WalterKolczynski-NOAA Please make this change:

```
diff --git a/ush/chgres_cube.sh b/ush/chgres_cube.sh
index 027b1e37..40d91f54 100755
--- a/ush/chgres_cube.sh
+++ b/ush/chgres_cube.sh
@@ -34,7 +34,7 @@ HOMEufs=${HOMEufs:-${NWROOT}/ufs_util.${ufs_ver}}
 EXECufs=${EXECufs:-$HOMEufs/exec}
 FIXufs=${FIXufs:-$HOMEufs/fix}
 FIXfv3=${FIXfv3:-$FIXufs/orog/C${CRES}}
-FIXsfc=${FIXsfc:-$FIXfv3/sfc}
+FIXsfc=${FIXsfc:-$FIXfv3/fix_sfc}
 FIXam=${FIXam:-$FIXufs/am}
```

**Comment by GeorgeGayno-NOAA** [comment]
@WalterKolczynski-NOAA And the jet directories need to be updated:

```
/lfs4/HFIP/emcda/George.Gayno/ufs_utils.git/UFS_UTILS.walter/fix $ git diff link_fixdirs.sh
diff --git a/fix/link_fixdirs.sh b/fix/link_fixdirs.sh
index ee19afa2..a236de8e 100755
--- a/fix/link_fixdirs.sh
+++ b/fix/link_fixdirs.sh
@@ -37,7 +37,7 @@ pwd=$(pwd -P)
 if [ $machine = "hera" ]; then
     FIX_DIR="/scratch1/NCEPDEV/global/glopara/fix"
 elif [ $machine = "jet" ]; then
-    FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix_NEW"
+    FIX_DIR="/lfs4/HFIP/hfv3gfs/glopara/git/fv3gfs/fix"
 elif [ $machine = "orion" ]; then
     FIX_DIR="/work/noaa/global/glopara/fix"
 elif [ $machine = "wcoss2" ]; then
```

**Comment by WalterKolczynski-NOAA** [comment]
@GeorgeGayno-NOAA Changes made. Unit tests and RT pass on Hera after the changes.

**Review by GeorgeGayno-NOAA — APPROVED** [review]
All regression tests now passing on all machines. Thanks, @WalterKolczynski-NOAA

**Pull request merged**